### PR TITLE
consider scrolling for hover function

### PR DESCRIPTION
**Pull request by mirnawong1**
this pr sets a flag to help indicate whether the page is being scrolled so that the toggles don’t expand while scrolling. this pr also modifies the `handleMouseEnter` function to check this flag and prevents the toggle from expanding if the page is being scrolled.
- adds a new state variable to track if the page is scrolling.
- add event listeners for scroll events to set this variable.
- modifies the handleMouseEnter function to check the scrolling state.
So:
- `isScrolling` tracks if user is scrolling
- `handleScroll` function to call on scrolling events and disables toggle if `isScrolling` is true
- `handleMouseEnter` checks for `isScrolling`

### CODE REVIEW

**Comment by mirnawong1** [comment]
preview of what this looks like: Recording Jan 11

**Review by matthewshaver — APPROVED** [review]
I'm still not sure that, for pages with a long list of items, the "hover to expand" is the best user experience, but this is a big improvement over how it behaved prior in regards to the scrolling. Thanks Mirna for your work on this!

File: website/docs/docs/dbt-cloud-apis/discovery-use-cases-and-examples.md [hunk 1]
```diff
@@ -7,13 +7,18 @@ With the Discovery API, you can query the metadata in dbt Cloud to learn more ab
...
+| [Quality](#quality) | Monitor data source freshness and test results to resolve issues<br /> and drive trust in data. | <ul><li>How fresh are my data sources?</li><li>Which tests and models failed?</li><li>What's my project's test coverage?</li></ul>  |
+| [Discovery](#discovery) | Find and understand relevant datasets and semantic nodes with<br /> rich context and metadata. | <ul><li>What do these tables and columns mean?</li><li>What's the full data lineage?</li><li>Which metrics can I query?</li> </ul> |
+| [Governance](#governance) | Audit data development and facilitate collaboration within and between teams. | <ul><li>Who is responsible for this model?</li><li>How do I contact the model's owner?</li><li>Who can use this model?</li></ul>|
+| [Development](#development) | Understand dataset changes and usage and gauge impacts<br /> to inform project definition. | <ul><li>How is this metric used in BI tools?</li><li>Which nodes depend on this data source?</li><li>How has a model changed? What impact?</li> </ul>|
+
+</table>
+</div>
```
Comment (JKarlavige) [inline website/docs/docs/dbt-cloud-apis/discovery-use-cases-and-examples.md]: Not sure if this edit is intended here but it looks like this table is broken within the preview: https://docs-getdbt-com-git-mwong-scroll-dbt-labs.vercel.app/docs/dbt-cloud-apis/discovery-use-cases-and-examples

**Review by JKarlavige — CHANGES_REQUESTED** [review]
The scroll feature looks like it works well! Just have one requested change regarding the table edit, then this should be ready to 🚢

File: website/docs/docs/dbt-cloud-apis/discovery-use-cases-and-examples.md (hunk 1 above)
Comment (mirnawong1) [reply website/docs/docs/dbt-cloud-apis/discovery-use-cases-and-examples.md]: oh wow thank you @JKarlavige ! that was definitely a mistaken and probably when i was playing with table widths. removed it and it should hopefully not affect the table anymore 🙏

**Comment by mirnawong1** [comment]
enabling auto merge

**Review by JKarlavige — APPROVED** [review]
:shipit: :shipit: :shipit:

**Pull request merged**